docs/chapters/traversal/index.md

```diff
@@ -0,0 +1,178 @@
+# Understand Jina Recursive Document Representation
+
+In Jina, each `Document` is represented as a recursive representation (tree).
+
+A rooted recursive representation has a root node and every node has X children.
+In Jina, the root node is the document itself, while the *left* & *right* children are referred to as *chunks* and *matches* respectively.
+
+![rooted-binary-tree](img/rooted-binary-tree.png)
```

Comment:
I am not sure a binary tree is the accurate description of what a document is, since the nodes are List themselves ... Any thoughts @maximilianwerk?

Reply:
It is definitely not a binary tree. It is just a tree with a root node. If you want to interpret the parent-child relationship as an ordering, it is an `ordered tree`.

Reply:
ok

Reply:
Let's avoid `binary-tree` naming also in the files

Reply:
updated